src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/toolkit/taglets/BaseTaglet.java

```diff
@@ -26,78 +26,66 @@
 package jdk.javadoc.internal.doclets.toolkit.taglets;
 
 import java.util.Set;
+
 import javax.lang.model.element.Element;
 
 import com.sun.source.doctree.DocTree;
 import com.sun.source.doctree.UnknownBlockTagTree;
+
 import jdk.javadoc.doclet.Taglet.Location;
+import jdk.javadoc.internal.doclets.toolkit.BaseConfiguration;
 import jdk.javadoc.internal.doclets.toolkit.Content;
+import jdk.javadoc.internal.doclets.toolkit.Messages;
+import jdk.javadoc.internal.doclets.toolkit.Resources;
+import jdk.javadoc.internal.doclets.toolkit.util.Utils;
 
 /**
  * A base class that implements the {@link Taglet} interface.
  */
 public class BaseTaglet implements Taglet {
+    // The following members are global to the lifetime of the doclet
+    protected final BaseConfiguration config;
+    protected final Messages messages;
+    protected final Resources resources;
+    protected final Utils utils;
 
+    // The following members are specific to the instance of the taglet
     protected final DocTree.Kind tagKind;
     protected final String name;
     private final boolean inline;
     private final Set<Location> sites;
 
-    BaseTaglet(DocTree.Kind tagKind, boolean inline, Set<Location> sites) {
-        this(tagKind.tagName, tagKind, inline, sites);
+    // The following is dynamically set for the duration of the methods
+    //      getInlineTagOutput and getAllBlockTagOutput
+    // by those taglets that need to refer to it
+    protected TagletWriter tagletWriter;
+
+    public BaseTaglet(BaseConfiguration config, DocTree.Kind tagKind, boolean inline, Set<Location> sites) {
```

Comment:
Good to see that the taglets now uniformly have useful state (BaseConfiguration). That reminded me of this [prior discussion of ours](https://github.com/openjdk/jdk/pull/10746/commits/1f24f1854d3aa93a77f002a2f2cf561959a98e2f#r1008348685).